packages/storage/amplify_storage_s3_dart/lib/src/model/s3_list_result.dart

```diff
@@ -27,12 +28,16 @@ class S3ListResult extends StorageListResult<S3Item> {
   ) {
     final output = paginatedResult.items;
     final metadata = S3ListMetadata.fromS3CommonPrefixes(
-      commonPrefixes: output.commonPrefixes?.toList(),
```

Comment:
`commonPrefixes` parameter was removed to reflect deprecation changes. It was previously used solely to derive subpath information, now deprecated and hoisted to `S3ListResult`